doc/model.rst

```diff
@@ -0,0 +1,683 @@
+In this example, we demonstrate how to instruct SmartSim to load
+an TensorFlow function into the database at model runtime. It's
+important to note the function, ``Model.add_function()`` is supported
+for non-colocated deployments and during a colocated deployment, ``Model.add_script()``
+should be used.
+
+**Python Script: Define a TF Function for Model Purposes**
+To load a TF function, define the function within the Python driver script.
+
+.. code-block:: python
+
+    def timestwo(x):
+        return 2*x
+
+**SmartSim Model Integration:**
+
+Assuming an initialized ``Model`` named `smartsim_model`, we specify the
+following parameters to the ``Model.add_function()`` function:
+
+.. code-block:: python
+
+    smartsim_model.add_function(name="example_func", function=timestwo, device="GPU", devices_per_node=2, first_device=0)
+
+In this integration, we provide the following details:
+
+-  `name`: "example_func" - A key to uniquely identify the model within the database.
+-  `function`: timestwo - Name of the TorchScript function defined in the Python driver script.
+-  `device`: "CPU" - Specifying the device for ML model execution.
+-  `devices_per_node`: 2 - Use two GPUs per node.
+-  `first_device`: 0 - Start with 0 index GPU.
+
+When the model is started via ``Experiment.start()``, the TF function will be loaded to the
+standard orchestrator that is launched prior to the start of the model.
+
+-------------------
+TorchScript Scripts
+-------------------
+SmartSim supports the execution of TorchScripts
+with Models. Regardless of whether
+the orchestrator is colocated or not, each script added to the Model is
+loaded into the orchestrator prior to the execution of entity.
+Users have the flexibility to choose
+between `"GPU"` or `"CPU"` for device selection, and in environments with multiple
+devices, specific device numbers can be specified via `devices_per_node` such as `"GPU:1,"`.
+
+When specifying a TorchScript using ``Model.add_script()``, the
+following arguments are offered:
+
+- name (str): key to store script under
+- script (str, optional): TorchScript code (only supported for non-colocated orchestrators)
+- script_path (str, optional): path to TorchScript code
+- device (str, optional): device for script execution, defaults to “CPU”
+- devices_per_node (int): The number of GPU devices available on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- first_device (int): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+
+You might use TorchScript scripts to represent individual models within the model.
+
+Example: Loading an TorchScript to the Model
+--------------------------------------------
+In this example, we demonstrate how to instruct SmartSim to load
+an TorchScript into the database at model runtime.
+
+**Python Script: Define a TorchScript for Model Purposes**
```

Comment:
Consider replacing "... for model purposes" with a more descriptive "Using Torch Scripts   As ML Models"